core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java

```diff
@@ -220,7 +284,49 @@ public void close() {
   @Override
   public Stream<PageInfo> getPages() throws IOException {
     Path rootDir = Paths.get(mRoot);
-    return Files.walk(rootDir).filter(Files::isRegularFile).map(this::getPageInfo);
+    return Files.walk(rootDir, FILE_LEVEL_DIR_DEPTH).filter(this::isFileLevelDir)
+        .flatMap(pathToFileDir -> {
+          try {
+            FileInfo fileInfo = loadFileInfo(pathToFileDir);
+            // remove all the stale sub-folder
+            // the timestamp of which is not equal to the last modified time in fileInfo
+            Files.list(pathToFileDir)
```

Comment:
It would be better to put the files.list into a try resource block to ensure the stream close after call list.

Reply:
Great! We have met this problem in our environment.It will lead to fd leak. https://stackoverflow.com/questions/36990053/resource-leak-in-files-listpath-dir-when-stream-is-not-explicitly-closed